pkg/services/k8s/apiserver/authorization/grafana_authorizer.go

```diff
@@ -0,0 +1,56 @@
+package authorization
+
+import (
+	"context"
+	"fmt"
+	"github.com/grafana/grafana/pkg/services/org"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+type GrafanaAuthorizer struct {
+}
+
+func NewGrafanaAuthorizer() *GrafanaAuthorizer {
+	return &GrafanaAuthorizer{}
+}
+
+func (auth GrafanaAuthorizer) Authorize(ctx context.Context, a authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
+	if userInfo, ok := request.UserFrom(ctx); ok {
```

Comment:
i'm finding it hard to track the nested if/else blocks. this is just a personal preference, but what do you think about something like:

```go
if userInfo, ok := request.UserFrom(ctx); !ok {
  return authorizer.DecisionDeny, fmt.Sprintf("user info missing in ctx, denied access on requested resource=%s, path=%s", a.GetResource(), a.GetPath()), nil
}

// pseudo-code - check k8s system admin
if group == "system:masters" {
   return authorizer.DecisionAllow, "", nil
}

// grafana authz logic here...
```

Reply:
+1 I'm a huge fan of structuring code so that the happy path reads top-to-bottom and exceptions go out to the right

Reply:
In other words, return early in failure cases as well. Yes, thanks for pointing it out.